third-party/realdds/include/realdds/dds-stream-server.h

```diff
@@ -186,4 +188,19 @@ class dds_pose_stream_server : public dds_motion_stream_server
 };
 
 
+class dds_metadata_stream_server : public dds_stream_server
+{
+    typedef dds_stream_server super;
+
+public:
+    dds_metadata_stream_server( std::string const & stream_name, std::string const & sensor_name );
+
+    char const * type_string() const override { return "metadata"; }
+
+    void open( std::string const & topic_name, std::shared_ptr< dds_publisher > const & ) override;
+
+    void publish( nlohmann::json && metadata );

```

Comment:
A non-virtual overloading (diff args) of the base-class virtual function?

Reply:
You are right. Will change as part of metadata not a dds_stream_server future change

Reply:
Added to the [to-do list](https://rsjira.intel.com/browse/LRS-585)

Reply:
No, this is a bug, need to fix it now...